### ISSUE TITLE
revertFunc misbehaves for events specified with the array form of resourceIds

### ISSUE THREAD

**Issue by kferrio** [fullcalendar/fullcalendar-scheduler#156]
#### UPDATE
**My initial analysis (below) was wrong. The problem has *nothing* to do with where the data are obtained. See elaboration further down in this thread.**

I read the rules at http://fullcalendar.io/wiki/Reporting-Bugs/ but I have an observation that occurs only when on pulling data from a JSON source.  I'd rather not stand up a public API to demo this, but maybe one already exists?

So... If I start with the example template for scheduler at JSBin *with static data*, everything renders as expected.  And if I add this
```
eventDrop: function (event, delta, revertFunc) { revertFunc(); }
```
then as expected every attempt to move an event is reverted.  All good.

If I instead pull the resources and events from an API
```
resources:
            {
                url: '/myapi/resources',
                error: function() {
                    $('#script-warning').text('Failed to load resources').show();
                }
            },
events:
           {
                url: '/myapi/events',
                error: function() {
                    $('#script-warning').text('Failed to load events').show();
                }
            },
```
all my events load and render as expected.  But then a strange thing happens when I try to move an event. Remember, every move will (should be) reverted. In fact all attempts to move an event horizontally (same resource, different time) are reverted. Good.  But if I try to move any given event vertically (same time, different resource OR different time, different resource) the *resource change is allowed!* The time change is reverted, as expected. Subsequent attempts to move the *same event* (in any direction) are reverted.  In other words, time changes are always disallowed as expected but resource changes are allowed on the first attempt (only) and reverted on all subsequent attempts for the same event.

To recap: when fetching events and resources from an API as JSON, the first call to revertFunc() appears to revert the time as expected but incorrectly permits the resource change (if requested).  All subsequent attempts to move the same event -- in any direction -- are rejected, as expected. This behavior is different from the behavior with static data.

I have not attempted to discern whether the problem is triggered only by pulling resources from an API, only by pulling events from an API, or only by pulling both from APIs.  In any event, my application needs to pull both from an API.

Also please note that all of my resource id's and event id's are integers. They do not start at 1 and they are not consecutive, but each is a unique key of course.

Please advise.

**Comment by kferrio** [comment]
Update: this may have something to do with the version of jQuery. Will investigate.

**Comment by kferrio** [comment]
The actual problem is resourceIds (array) versus resourceId.  
See  [this JSBin](http://jsbin.com/nenoqu/edit?js,output) for a complete demo.

An event for which resource(s) are specified using the array form, with key `resourceIds` is not handled correctly by the `revertFunc` provided to the `eventDrop` handler.  An event specified in this way, regardless of the actual number of resources it references, will have its time but not it's resource(s) reverted one the first attempt to revert a drag-and-drop action. All subsequent attempts to revert a drag-and-drop on the same object will succeed, *viz.* reverting the the requested change works as expected.

Please do not be distracted (much) by the fact that dragging an event really makes sense only along the time axis. For simplicity, in the JSBin, focus just on the the two events with one resource each, and ignore the event with two resources. That's another issue.

**Comment by kferrio** [comment]
I've briefly reviewed the Calendar.Extensions in `scheduler.js` and the mouse-down code in `fullcalendar.js` -- where the drag-and-drop starts -- and I don't see a quick win without side-effects. Isn't it remarkable how many edge-cases come up when we're talking about time?

**Comment by arshaw** [comment]
im having a hard time parsing through all this to figure out what the core issue is. could be so kind as to write a summary of the problem, as well as what to do when I open up the JSBin to recreate?

**Comment by kferrio** [comment]
It appears that I may have inadvertently edited the JSBin a day or two after opening this issue. I'm out of the office today and will correct that when i return. JSBin seems not to work with my phone's browser.

In brief, if an event is defined using the resourceIds (array) property and that event is dragged, then calling revertFunc reverts the time change but does *not* revert the resource change. This error happens *only* on the first drag. After the first drag, reverts of subsequent drags behave as expected...*both* time change and resource change are reverted. As my corrected JSBin will show, the error occurs even if the rwsourceId array has our one elelent.

If an event is specified with the scalar resourceId then there is no problem.

For anyone experiencing this problem, here is a workaround:
Do not use the array resourceIds property of event. Use the scalar resourceId. So specify an event with N resources as N separate events, and when a drag occurs, do two things:
1. Post the 1 updated event to the server, which the server may accept or reject based on its knowledge of the other N-1 resources' availability, and
2. invoke a refetch to reflect the result

In other words, until reverting works reliably for events specified with a resourceIds array, we need to do our own overlap checking for multi-resource events. I chose to do the check with my existing server-side validation code. We could also interrogate the fullcalendar object of course if we trusted the client and that would allow for prettier revert animation..

**Comment by kferrio** [comment]
Restored the correct [JSBin](http://jsbin.com/nenoqu/edit?js,output)

Please read the comments on the events in the JSBin. The easiest way to see what is going on is to try to drag-and-drop diagonally.

**Comment by arshaw** [comment]
Sorry for the wait. Thanks for the well-created JSBin; I was able to confirm this.

**Comment by arshaw** [comment]
fix release in [v1.3.2](https://github.com/fullcalendar/fullcalendar-scheduler/releases/tag/v1.3.2)

**Comment by kferrio** [comment]
Yeah! Awesome!
